tests/qtrajectory_test.cc

```diff
@@ -0,0 +1,439 @@
+       {normal, 0, p2, {M::Create(time, 0, {0, 0, s, 0, 0, 0, 0, 0})}}});
+  ncircuit.push_back(
+      {{normal, 0, p1, {M::Create(time, 1, {1, 0, 0, 0, 0, 0, r, 0})}},
+       {normal, 0, p2, {M::Create(time, 1, {0, 0, s, 0, 0, 0, 0, 0})}}});
+}
+
+void AddGenAmplDumpNoise(
+    unsigned time, double g, types::NoisyCircuit& ncircuit) {
+  using fp_type = types::Gate::fp_type;
+
+  // Probability of exchanging energy with the environment.
+  double p = 0.5;
+
+  double p1 = p * (1 - g);
+  double p2 = (1 - p) * (1 - g);
+  double p3 = 0;
+
+  fp_type t1 = std::sqrt(p);
+  fp_type r1 = std::sqrt(p * (1 - g));
+  fp_type s1 = std::sqrt(p * g);
+  fp_type t2 = std::sqrt(1 - p);
+  fp_type r2 = std::sqrt((1 - p) * (1 - g));
+  fp_type s2 = std::sqrt((1 - p) * g);
+
+  auto normal = KrausOperator<types::Gate>::kNormal;
+
+  using M = Cirq::MatrixGate1<fp_type>;
+
+  ncircuit.push_back(
+      {{normal, 0, p1, {M::Create(time, 0, {t1, 0, 0, 0, 0, 0, r1, 0})}},
+       {normal, 0, p2, {M::Create(time, 0, {r2, 0, 0, 0, 0, 0, t2, 0})}},
+       {normal, 0, p3, {M::Create(time, 0, {0, 0, s1, 0, 0, 0, 0, 0})}},
+       {normal, 0, p3, {M::Create(time, 0, {0, 0, 0, 0, s2, 0, 0, 0})}}});
+  ncircuit.push_back(
+      {{normal, 0, p1, {M::Create(time, 1, {t1, 0, 0, 0, 0, 0, r1, 0})}},
+       {normal, 0, p2, {M::Create(time, 1, {r2, 0, 0, 0, 0, 0, t2, 0})}},
+       {normal, 0, p3, {M::Create(time, 1, {0, 0, s1, 0, 0, 0, 0, 0})}},
+       {normal, 0, p3, {M::Create(time, 1, {0, 0, 0, 0, s2, 0, 0, 0})}}});
+}
+
+template <typename AddNoise>
+types::NoisyCircuit GenerateNoisyCirquit(double p, AddNoise&& add_noise) {
+  using fp_type = types::Gate::fp_type;
+
+  types::NoisyCircuit ncircuit;
+  ncircuit.reserve(24);
+
+  using Hd = Cirq::H<fp_type>;
+  using IS = Cirq::ISWAP<fp_type>;
+  using Rx = Cirq::rx<fp_type>;
+  using Ry = Cirq::ry<fp_type>;
+
+  auto normal = KrausOperator<types::Gate>::kNormal;
+
+  ncircuit.push_back({{normal, 1, 1.0, {Hd::Create(0, 0)}}});
+  ncircuit.push_back({{normal, 1, 1.0, {Hd::Create(0, 1)}}});
+  add_noise(1, p, ncircuit);
+  ncircuit.push_back({{normal, 1, 1.0, {IS::Create(2, 0, 1)}}});
+  add_noise(3, p, ncircuit);
+  ncircuit.push_back({{normal, 1, 1.0, {Rx::Create(4, 0, 0.7)}}});
+  ncircuit.push_back({{normal, 1, 1.0, {Ry::Create(4, 1, 0.1)}}});
+  add_noise(5, p, ncircuit);
+  ncircuit.push_back({{normal, 1, 1.0, {IS::Create(6, 0, 1)}}});
+  add_noise(7, p, ncircuit);
+  ncircuit.push_back({{normal, 1, 1.0, {Ry::Create(8, 0, 0.4)}}});
+  ncircuit.push_back({{normal, 1, 1.0, {Rx::Create(8, 1, 0.7)}}});
+  add_noise(9, p, ncircuit);
+  ncircuit.push_back({{normal, 1, 1.0, {IS::Create(10, 0, 1)}}});
+  add_noise(11, p, ncircuit);
+  ncircuit.push_back({{KrausOperator<types::Gate>::kMeasurement, 1, 1.0,
+                       {gate::Measurement<types::Gate>::Create(12, {0, 1})}}});
+  add_noise(13, p, ncircuit);
+
+  return ncircuit;
+}
+
+void Run1(const types::NoisyCircuit& ncircuit,
```

Comment:
Is the only distinction between `Run1` and `Run2` that `Run1` calls the run-multiple method once, while `Run2` calls the run-once method multiple times? If so, it might be useful to name these `RunBatch` and `RunOnceRepeatedly` (or something along those lines).

Reply:
Done.